### PR TITLE
Add explicit conversions to the base type for type and metadata types

### PR DESCRIPTION
**Pull request by dkrautha**

### CODE REVIEW

File: xdl_rs/src/conversions/base_type.rs [hunk 1]
```diff
@@ -11,18 +11,39 @@ impl From<XdlPrimitiveMetadata> for XdlMetadata {
     }
 }
 
+// TODO: Test this
```
Comment (ahuston-0) [inline xdl_rs/src/conversions/base_type.rs:14]: Do you you want me to merge this before test cases are in? Not sure how hard it would be to test this.

Comment (ahuston-0) [reply xdl_rs/src/conversions/base_type.rs:14]: If you want me to merge now and then you can do this later, all good. Just wanted to get confirmation before I hit the button.

**Review by ahuston-0 — CHANGES_REQUESTED** [review]
Pushing back for review

File: xdl_rs/src/conversions/base_type.rs (hunk 1 above)
Comment (ahuston-0) [reply xdl_rs/src/conversions/base_type.rs:14]: Yeah i think that should work, as long as you're validating state after (and/or before?). Actually using the stuff is a good way to do unit testing i think, since that can catch weird edge cases that the monotonous ones don't.

Comment (dkrautha) [reply xdl_rs/src/conversions/base_type.rs]: Added tests for the stuff we talked about (except for struct, as its not done yet). I also did some wacky trait things that I think you'll enjoy :)
